lib/pageflow/global_config_api.rb

```diff
@@ -44,8 +44,13 @@ def configure(&block)
     # @return [Configuration]
     # @since 0.9
     def config_for(target)
-      build_config do |config|
-        config.enable_features(target.enabled_feature_names)
+      config = build_config do |c|
+        c.enable_features(target.enabled_feature_names)
+      end
+      if target.respond_to?(:type_name) && config.entry_type_configs.key?(target.type_name)
+        ConfigDelegator.new(config, target.type_name)
+      else
+        config
```

Comment:
The type of the return value of `config_for` should not depend on whether `for_entry_type` was called. Otherwise it's not safe to use entry type specific methods. This could also be spec.